src/Controller/Entry/EntryCreateController.php

```
@@ -40,17 +43,55 @@ public function __construct(
         private readonly TagRepository $tagRepository,
         private readonly EntryManager $manager,
         private readonly EntryCommentManager $commentManager,
+        private readonly ImageRepository $imageRepository,
+        private readonly ImageFactory $imageFactory,
         private readonly ValidatorInterface $validator,
         private readonly IpResolver $ipResolver,
         private readonly Security $security,
     ) {
     }
 
+    /**
+     * @param string[]|null $tags
+     */
     #[IsGranted('ROLE_USER')]
-    public function __invoke(?Magazine $magazine, Request $request): Response
-    {
+    public function __invoke(
+        ?Magazine $magazine,
+        #[MapQueryParameter]
+        ?string $title,
+        #[MapQueryParameter]
+        ?string $url,
+        #[MapQueryParameter]
+        ?string $body,
+        #[MapQueryParameter]
+        ?string $imageAlt,
+        #[MapQueryParameter]
+        ?string $isNsfw,
+        #[MapQueryParameter]
+        ?string $isOc,
+        #[MapQueryParameter]
+        ?array $tags,
+        #[MapQueryParameter]
+        ?string $imageHash,
+        Request $request,
+    ): Response {
         $dto = new EntryDto();
         $dto->magazine = $magazine;
+        $dto->title = $title;
+        $dto->url = $url;
+        $dto->body = $body;
+        $dto->imageAlt = $imageAlt;
+        $dto->isAdult = '1' === $isNsfw;
+        $dto->isOc = '1' === $isOc;
+        $dto->tags = $tags;
+
+        if (null !== $imageHash) {
+            $img = $this->imageRepository->findOneBySha256(hex2bin($imageHash));
+            if (null !== $img) {
```

can `null === $img` be silently ignored or should I throw an exception?